internal/access/credential.go

```diff
@@ -23,11 +24,26 @@ func CreateCredential(c *gin.Context, user models.Identity) (string, error) {
 		return "", HandleAuthErr(err, "user", "create", models.InfraAdminRole)
 	}
 
-	tmpPassword, err := generate.CryptoRandom(12, generate.CharsetPassword)
+	settings, err := data.GetSettings(db)
+	if err != nil {
+		return "", err
+	}
+
+GENERATE:
```

Comment:
This is problematic if the user sets a password requirement that is hard for the generator to generate, e.g. high minimum of lowercase characters, and potentially create an infinite loop if the requirements are specific enough.

The problem is that the generator charset is not synced to the password requirement charset. You could synchronize the charsets but I feel like validating randomly generated values is unnecessary.